rhttpc-transport/src/main/scala/rhttpc/transport/Message.scala

```diff
@@ -37,7 +37,17 @@ object DelayedMessage {
         val delay = props(MessagePropertiesNaming.delayProperty).asInstanceOf[Number].longValue() millis
         val attempt = props.get(MessagePropertiesNaming.attemptProperty).map(_.asInstanceOf[Number].intValue()).getOrElse(1)
         // It might fail without getOrElse for messages of older formats without that field
-        val firstAttemptTimestamp = props.get(MessagePropertiesNaming.firstAttemptDate).map(_.asInstanceOf[String]).map(Instant.parse).getOrElse(Instant.now())
+        val firstAttemptTimestamp = props.get(MessagePropertiesNaming.firstAttemptTimestamp)
+          .map(_.asInstanceOf[Number].longValue())
+          .map(Instant.ofEpochMilli)
+          .getOrElse(Instant.now())
+
+        // RabbitMq for some reason serializes and deserializes String as a LongString (see ValueReader and ValueWriter)
```

Comment:
This comment in this place can be confusing now. WDYT about moving it to lets say `SendingFullMessage.prepareMessage` or `AmqpTransportImpl` and reformat a bit, that it will be a warning in case if you want to use headers?

Reply:
Moved this comment above SendingFullMessage.prepareMessage and added a warning above DelayedMessage object to point to that class